pkg/operator/controller/garden/garden/reconciler_reconcile.go

```diff
@@ -275,7 +308,7 @@ func (r *Reconciler) reconcile(
 			}).
 				RetryUntilTimeout(5*time.Second, 30*time.Second).
 				DoIf(helper.GetServiceAccountKeyRotationPhase(garden.Status.Credentials) == gardencorev1beta1.RotationPreparing),
-			Dependencies: flow.NewTaskIDs(deployVirtualGardenGardenerAccess),
+			Dependencies: flow.NewTaskIDs(deployVirtualGardenGardenerAccess, deployGardenerAPIServer, deployGardenerAdmissionController, deployGardenerControllerManager, deployGardenerScheduler),
```

Comment:
Should we also the a dependency to the `Kube-Controller-Manager`?

Reply:
Good question - I think it's even better to do what we already do in the shoot controller: https://github.com/gardener/gardener/blob/6f8ce7692f793e583734f533cfdee7b9d93fb815/pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go#L314
Thanks for pointing this out!

Reply:
It seems that this (ref https://github.com/gardener/gardener/pull/8309/commits/67ab1d05c1687f635de71ac61f933f942361af01) leads to failing e2e tests ([example 1](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/8309/pull-gardener-e2e-kind-operator/1697521409309282304), [example 2](https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/8309/pull-gardener-e2e-kind-operator/1697531581914157056)), so I reverted the change and follow what you suggested initially. I'm not sure what the exact difference is in the shoot controller, but I think it's okay for now.